test/junit/scala/collection/immutable/TreeSetTest.scala

```diff
@@ -213,4 +213,33 @@ class TreeSetTest extends AllocationTest{
       assertEquals(s"$rText $lText", rhs, lhs)
     }
   }
+
+  @Test
+  def unionAndIntersectRetainLeft(): Unit = {
```

Comment:
looks like this is a SetTest, not a TreeSetTest

sort of 
```
for ( lhs <- anySetType;
        rhs <- andSetType;
        values <- testValues) {
   assertIdenticalElements(...)
```

Reply:
The 2.13.x branch contains this sort of test but I wanted to keep the backport smaller and just backport concrete tests for the bug.